### PR TITLE
Pub 1310 Create summary text for emails

### PR DESCRIPTION
**Pull request by furnivall**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1310

### Change description ###
Added methods to produce email summaries for the SJP Press, SJP Public and Civil Daily Cause List types. When an email is compiled from one of these types, a pleasing summary will appear at the bottom of the email for the user to enjoy.

I genuinely apologise for how awful this code looks - feel free to refactor if it would make you happy, but I promise I've done what I can. Json just kind of sucks to parse.

Also suppressed a silly error CVE.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+     * @param sessionChannel - session channel passed in from parent method - overridden if sitting level channel
+     *                       exists.
+     * @return string of these bits.
+     */
+    private String processCivilDailySittings(JsonNode node, List<String> sessionChannel) {
+        JsonNode sittingNode = node.get(SESSION).get(0).get(SITTINGS);
+        Iterator<JsonNode> sittingIterator = sittingNode.elements();
+        StringBuilder outputString = new StringBuilder(26);
+        int counter = 1;
+        boolean sorryPmd = sittingNode.size() > 1;
```
Comment (joshblackmoor) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java:147]: Can this boolean name be changed from sorryPmd?

**Review by joshblackmoor — CHANGES_REQUESTED** [review]
Just one variable name change. Have seen a demo from you and looks good :)

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        StringBuilder outputString = new StringBuilder();
+        List<String> sessionChannel;
+        TypeReference<List<String>> typeReference = new TypeReference<>() {
+        };
+        while (courtRoomNode.hasNext()) {
+            JsonNode thisCourtRoom = courtRoomNode.next();
+            JsonNode sessionChannelNode = thisCourtRoom.get(SESSION).get(0).path("sessionChannel");
+            sessionChannel = mapper.convertValue(sessionChannelNode, typeReference);
+            outputString.append("\n\nCourtroom: ").append(thisCourtRoom.get("courtRoomName").asText());
+            outputString.append(processCivilDailySittings(thisCourtRoom, sessionChannel));
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java:99]: You are using the fluent style to append to outputString in the previous line. It is not on this and the next line. May be we should keep it consistent and using the fluent '.append()' here too.

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        StringBuilder johName = new StringBuilder("\nJudiciary: ");
+        int counter = 1;
+        int length = judiciaryNode.size();
+        while (johNode.hasNext()) {
+            JsonNode currentJoh = johNode.next();
+            String title = currentJoh.path("johTitle").asText();
+            String knownAs = currentJoh.path("johKnownAs").asText();
+            johName.append(title).append(' ');
+            johName.append(knownAs);
+            if (counter < length) {
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java:126]: Think we can just use "if (johNode.hasNext())" as the condition here and do not need "length" and "counter"?

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java (hunk 1 above)
Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java]: done, but begrudgingly

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    /**
+     * offences iterator method - handles logic of accused of single or multiple offences and returns output string.
+     *
+     * @param offencesNode - iterator on offences.
+     * @return string with offence data.
+     */
+    private String processOffencessjpPress(JsonNode offencesNode) {
+        StringBuilder outputString = new StringBuilder();
+        boolean sorryPmd = offencesNode.size() > 1;
+        if (sorryPmd) {
```
Comment (joshblackmoor) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java:244]: boolean name change here

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java (hunk 1 above)
Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java]: May be better to use a more descriptive name

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java (hunk 3 above)
Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java]: man this was a really silly mistake. I even used johNode.hasNext() as the loop condition. Silly danny.

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java (hunk 2 above)
Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java]: fixed this now.

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java (hunk 4 above)
Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java]: sorted

**Review by joshblackmoor — APPROVED** [review]
Looks good 👍

**Review by KianKwa — APPROVED** [review]

File: config/owasp/suppressions.xml [hunk 5]
```diff
@@ -47,4 +47,9 @@
     <packageUrl regex="true">^pkg:maven/org.apache.logging.log4j/log4j-api@2.17.2</packageUrl>
     <cve>CVE-2022-33915</cve>
   </suppress>
+  <suppress>
```
Comment (ChrisS1512) [inline config/owasp/suppressions.xml:50]: Wondering if this is still needed as the master build ran yesterday. Could you try removing it and see if it still passes?

File: src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+     * handles reporting restrictions for sjp press.
+     *
+     * @param node - node which is checked for reporting restriction.
+     * @return - text based on whether restriction exists.
+     */
+    private String processReportingRestrictionsjpPress(JsonNode node) {
+        boolean restriction = node.get("reportingRestriction").asBoolean();
+        if (restriction) {
+            return "(Reporting restriction)";
+        } else {
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/publication/services/service/ArtefactSummaryService.java:270]: Don't need the extra else here :)

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**